scheduler/server/stateful_scheduler.go

```diff
@@ -292,7 +297,15 @@ func NewStatefulScheduler(
 		stat:             stat,
 
 		tasksByJobClassAndStartTimeSec: tasksByClassAndStartMap,
+		persistor:                      persistor,
+	}
+
+	sched.setThrottle(-1)
+
+	if sched.persistor == nil {
+		sched.persistor = Persistor(&nopPersistor{})
 	}
+	sched.persistor.LoadSettings(sched)
```

Comment:
does this override the setThrottle above?

Reply:
It depends on the implementation of LoadSettings()...the nop persistor doesn't do anything so throttle is not changed.
The Persistor implemented in our twitter only code (source) behaves as follows:
- don't do anything if no setting have been saved
- if any setting is changed, all settings are persisted so LoadSetting will set the correct values.